### DISCUSSION TITLE
App health with CRD implementing cronjob

### DISCUSSION THREAD

**Discussion by mtotone** [argoproj/argo-cd#20536] [General]
Hello,

I try to manage mariadb-operator resources with an internal helm app and we're facing a technical issue with backup CRD.

Backup CRD is a cronjob and the app health is shown as progressing until cronjob is executed for the first time.

Healthy app:
`NAMESPACE  NAME                     COMPLETE   STATUS              MARIADB                              AGE
<namespace>  backup-scheduled   True               Success             <cluster>   22h`

Progressing app:
`NAMESPACE  NAME                     COMPLETE   STATUS      MARIADB                              AGE
<namespace>  backup-scheduled   False              Scheduled   <cluster>             6m17s`

For the progressing app, the cronjob last schedule is none.

For now, i use a lua script to manage mariadb-operator backup object and set status when reason is 'CronJobScheduled'.

Have you ever face a similar issue (with mariadb-operator or other) and how did you solve it ?

Thanks.

**Comment by HanssenKai** [comment]
Unless someone contributes a custom health check for mariaDB resources to the argoCD repo, then a lua override is the way to go. In the meantime, it would be nice if you could share the lua snippet for others who run in to the same issue.  

By default, kubernetes cronJobs have empty status fields until they are run the first time, so a mariaDB resource health check would probably need an additional check for `status.type == Complete` and filter by `status.reason == CronJobSuccess | CronJobFailed | CronJobScheduled`
